test/model/tools/model_test/main.cpp

```diff
@@ -98,13 +99,27 @@ run_and_verify(std::shared_ptr<model::kv_workload> workload, const std::string &
         throw std::runtime_error("Failed to close file: " + workload_file);
 
     /* Run the workload in WiredTiger. */
+
+    std::vector<int> ret_wt;
     try {
-        workload->run_in_wiredtiger(home.c_str(), conn_config.c_str(), table_config.c_str());
+        ret_wt =
+          workload->run_in_wiredtiger(home.c_str(), conn_config.c_str(), table_config.c_str());
     } catch (std::exception &e) {
         throw std::runtime_error(
           "Failed to run the workload in WiredTiger: " + std::string(e.what()));
     }
 
+    /* Compare the return codes. */
+    size_t min_ret_length = std::min(ret_model.size(), ret_wt.size());
+    for (size_t i = 0; i < min_ret_length; i++)
+        if (ret_model[i] != ret_wt[i])
+            throw std::runtime_error("Return codes differ for operation " + std::to_string(i + 1) +
+              ": WiredTiger returned " + std::to_string(ret_wt[i]) + ", but " +
+              std::to_string(ret_model[i]) + " was expected.");
+    if (ret_model.size() != ret_wt.size())
```

Comment:
Is it worth asserting that the lists aren't both zero length? Here and in `verify_workload`.

Reply:
Strictly speaking, it is not illegal for the workload to not contain any operations, so we can't assert the list to not be zero-length.

Reply:
Fair enough - thanks for the explanation.